### DISCUSSION TITLE
`remark-rehype`: Custom handlers

### DISCUSSION THREAD

**Discussion by jirpok** [unifiedjs/unified#249] [Q&A] [answered]
Hey guys,

I'm struggling to understand how handlers work. 

I've been playing around with some simple ideas, such as adding/modifying properties, but I keep getting various errors. 

```js
.use(remarkRehype, {
    handlers: {
        link: (state, node, parent) => {
            // ?
        }
    }
})
```

Furthermore, when a node has children, and I modify that node with a handler, how do I properly reattach the child nodes?

One real world example would be removing/not including the `align` attribute on table cells (it's just an example, I know it is recommended to use a package).

I haven't found much on the topic besides what's in the README, and #21, which got me started at least. I've also checked out the default handlers in `mdast-util-to-hast/lib/handlers`.

**Comment by wooorm** [comment, marked as answer]
Hey!

To change nodes, don’t use `remark-rehype` handlers. Those functions map one mdast node to another hast node. Changing nodes is better done in other tools. How that happens is pretty strongly defined by markdown/GFM. But handlers mostly exist for custom nodes.

I’d recommend making a rehype plugin if you want to change nodes representing HTML elements. That’s the proper place to remove/change/add properties. See https://unifiedjs.com/learn/guide/create-a-rehype-plugin/ for a guide.
